### PR TITLE
新增工具接口

### PR DESCRIPTION
**Pull request by leej1012**

### CODE REVIEW

File: src/main/java/com/ontology/controller/ToolsController.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        }
+
+        String decMsg = toolsService.decrypt(action,ontid,ddo,kid,password,secStr);
+
+        return new Result(action, ErrorInfo.SUCCESS.code(), ErrorInfo.SUCCESS.descEN(), decMsg);
+    }
+
+
+    private void helpCheckPwd(String action, String pwd) {
+        if (pwd == null || pwd.length() < 8 || pwd.length() > 15) {
```
Comment (NashMiao) [inline src/main/java/com/ontology/controller/ToolsController.java:83]: Why the length of `pwd` should from 8 to 15?

File: src/main/java/com/ontology/service/impl/BuyerServiceImpl.java [hunk 2]
```diff
@@ -293,23 +269,23 @@ public void run() {
...
-                msg[i] = (String) objects[i];
-            }
-            byte[] decrypt = ECIES.Decrypt(buyerAcct, msg);
-            message.add(new String(decrypt,"utf-8").replace("\"",""));
-        }
-        log.info("{}",message);
-        return message;
-    }
+//    @Override
+//    public List<String> decodeMessage(String action, String dataDemander, String password, List<String> secStr) throws Exception {
```
Comment (NashMiao) [inline src/main/java/com/ontology/service/impl/BuyerServiceImpl.java]: Maybe you need to check whether there is annotation or not.

File: src/main/java/com/ontology/service/impl/SellerServiceImpl.java [hunk 3]
```diff
@@ -68,8 +68,9 @@ public void deliverData(String action, String dataProvider, String password, Str
         List<String> dataIdList = new ArrayList<>();
         for (Object o : encMsgList) {
             if(o instanceof String) {
```
Comment (NashMiao) [inline src/main/java/com/ontology/service/impl/SellerServiceImpl.java:64]: maybe you need to add `else` in here.

File: src/main/java/com/ontology/service/impl/ToolsServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    }
+
+    @Override
+    public String decrypt(String action, String ontid, String ddo, Integer kid, String password, String secStr) throws Exception {
+        String publicKey = getPublicKey(action,ddo,kid);
+
+        OntId OntId = getOntId(action, ontid, password);
+        JSONObject jsonObject = JSONObject.parseObject(OntId.getKeystore());
+        if (publicKey.equals(jsonObject.getString("publicKey"))) {
+            Account buyerAcct = sdk.getAccount(OntId.getKeystore(), password);
```
Comment (NashMiao) [inline src/main/java/com/ontology/service/impl/ToolsServiceImpl.java]: Maybe you should change it's name to `acct`. In general, anyone with `ONT ID` can use this interface.

```java
Account acct = sdk.getAccount(OntId.getKeystore(), password);
```

File: src/main/java/com/ontology/controller/ToolsController.java (hunk 1 above)
Comment (leej1012) [reply src/main/java/com/ontology/controller/ToolsController.java:83]: 密码格式目前是沿袭OntId项目

File: src/main/java/com/ontology/service/impl/BuyerServiceImpl.java (hunk 2 above)
Comment (leej1012) [reply src/main/java/com/ontology/service/impl/BuyerServiceImpl.java]: https://github.com/ontio-ddxf/data-dealer/pull/7/commits/f922720a7aefb4bc27652a8405f35092b4a9fa5b

File: src/main/java/com/ontology/service/impl/SellerServiceImpl.java (hunk 3 above)
Comment (leej1012) [reply src/main/java/com/ontology/service/impl/SellerServiceImpl.java:64]: https://github.com/ontio-ddxf/data-dealer/pull/7/commits/f922720a7aefb4bc27652a8405f35092b4a9fa5b

File: src/main/java/com/ontology/service/impl/ToolsServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        if (0==provider) {
+            // 需求方
+            queryType = "buyer_ontid";
+        } else if (1==provider) {
+            // 提供方
+            queryType = "seller_ontid";
+        } else {
+            throw new OntIdException(action, ErrorInfo.PARAM_ERROR.descCN(), ErrorInfo.PARAM_ERROR.descEN(), ErrorInfo.PARAM_ERROR.code());
+        }
+        List<Order> orderList = orderMapper.getOrderList(queryType,buyerOntid);
```
Comment (NashMiao) [inline src/main/java/com/ontology/service/impl/ToolsServiceImpl.java]: Maybe you should change `buyerOntid` to `ontid`. What I understand in here is both `buyer` and `seller` can use this interface.

File: src/main/java/com/ontology/service/impl/BuyerServiceImpl.java [hunk 6]
```diff
@@ -112,7 +111,6 @@ public void run() {
                     Thread.sleep(6*1000);
                     Object event = sdk.checkEvent(txHash);
                     while (Helper.isEmptyOrNull(event)) {
-//                        sdk.invokeContract(params,buyerAcct, payerAcct,false);
                         Thread.sleep(6*1000);
```
Comment (NashMiao) [inline src/main/java/com/ontology/service/impl/BuyerServiceImpl.java]: I think we need to set a upper limit for  executing `checkEvent`.

File: src/main/java/com/ontology/service/impl/ToolsServiceImpl.java (hunk 5 above)
Comment (leej1012) [reply src/main/java/com/ontology/service/impl/ToolsServiceImpl.java]: https://github.com/ontio-ddxf/data-dealer/pull/7/commits/f922720a7aefb4bc27652a8405f35092b4a9fa5b

File: src/main/java/com/ontology/service/impl/BuyerServiceImpl.java [hunk 7]
```diff
@@ -177,7 +175,6 @@ public void run() {
...
                         event = sdk.checkEvent(txHash);
```
Comment (NashMiao) [inline src/main/java/com/ontology/service/impl/BuyerServiceImpl.java:186]: I think we need to set a upper limit for  executing `checkEvent`.

File: src/main/java/com/ontology/service/impl/BuyerServiceImpl.java [hunk 8]
```diff
@@ -274,7 +247,6 @@ public void run() {
...
```
Comment (NashMiao) [inline src/main/java/com/ontology/service/impl/BuyerServiceImpl.java:264]: I think we need to set a upper limit for  executing `checkEvent`.

File: src/main/java/com/ontology/service/impl/ToolsServiceImpl.java (hunk 4 above)
Comment (leej1012) [reply src/main/java/com/ontology/service/impl/ToolsServiceImpl.java]: https://github.com/ontio-ddxf/data-dealer/pull/7/commits/f922720a7aefb4bc27652a8405f35092b4a9fa5b

File: src/main/java/com/ontology/service/impl/OntIdServiceImpl.java [hunk 9]
```diff
@@ -160,7 +160,6 @@ public void run() {
                     Thread.sleep(6*1000);
                     Object event = sdk.checkEvent(txHash);
                     while (Helper.isEmptyOrNull(event)) {
-//                        sdk.addAttributes(ontId,password,key,valueType,value);
                         Thread.sleep(6*1000);
                         event = sdk.checkEvent(txHash);
```
Comment (NashMiao) [inline src/main/java/com/ontology/service/impl/OntIdServiceImpl.java:165]: I think we need to set a upper limit for  executing `checkEvent`.

File: src/main/java/com/ontology/service/impl/SellerServiceImpl.java [hunk 10]
```diff
@@ -92,7 +87,6 @@ public void run() {
                     Thread.sleep(6*1000);
                     Object event = sdk.checkEvent(txHash);
                     while (Helper.isEmptyOrNull(event)) {
-//                        sdk.invokeContract(params,sellerAcct, payerAcct,false);
                         Thread.sleep(6*1000);
                         event = sdk.checkEvent(txHash);
```
Comment (NashMiao) [inline src/main/java/com/ontology/service/impl/SellerServiceImpl.java:92]: I think we need to set a upper limit for  executing `checkEvent`.

File: src/main/java/com/ontology/service/impl/SellerServiceImpl.java [hunk 11]
```diff
@@ -146,7 +140,6 @@ public void run() {
...
```
Comment (NashMiao) [inline src/main/java/com/ontology/service/impl/SellerServiceImpl.java:151]: I think we need to set a upper limit for  executing `checkEvent`.

File: src/main/java/com/ontology/service/impl/BuyerServiceImpl.java (hunk 6 above)
Comment (leej1012) [reply src/main/java/com/ontology/service/impl/BuyerServiceImpl.java]: https://github.com/ontio-ddxf/data-dealer/pull/7/commits/f922720a7aefb4bc27652a8405f35092b4a9fa5b

File: src/main/java/com/ontology/service/impl/BuyerServiceImpl.java (hunk 7 above)
Comment (leej1012) [reply src/main/java/com/ontology/service/impl/BuyerServiceImpl.java:186]: https://github.com/ontio-ddxf/data-dealer/pull/7/commits/f922720a7aefb4bc27652a8405f35092b4a9fa5b

File: src/main/java/com/ontology/service/impl/BuyerServiceImpl.java (hunk 8 above)
Comment (leej1012) [reply src/main/java/com/ontology/service/impl/BuyerServiceImpl.java:264]: https://github.com/ontio-ddxf/data-dealer/pull/7/commits/f922720a7aefb4bc27652a8405f35092b4a9fa5b

File: src/main/java/com/ontology/service/impl/OntIdServiceImpl.java (hunk 9 above)
Comment (leej1012) [reply src/main/java/com/ontology/service/impl/OntIdServiceImpl.java:165]: https://github.com/ontio-ddxf/data-dealer/pull/7/commits/f922720a7aefb4bc27652a8405f35092b4a9fa5b

File: src/main/java/com/ontology/service/impl/SellerServiceImpl.java (hunk 10 above)
Comment (leej1012) [reply src/main/java/com/ontology/service/impl/SellerServiceImpl.java:92]: https://github.com/ontio-ddxf/data-dealer/pull/7/commits/f922720a7aefb4bc27652a8405f35092b4a9fa5b

File: src/main/java/com/ontology/service/impl/SellerServiceImpl.java (hunk 11 above)
Comment (leej1012) [reply src/main/java/com/ontology/service/impl/SellerServiceImpl.java:151]: https://github.com/ontio-ddxf/data-dealer/pull/7/commits/f922720a7aefb4bc27652a8405f35092b4a9fa5b

**Review by NashMiao — APPROVED** [review]

**Pull request merged**